nack/receive_log.go

```diff
@@ -0,0 +1,141 @@
+package nack
+
+import (
+	"errors"
+	"strconv"
+	"sync"
+)
+
+var (
+	allowedReceiveLogSizes     map[uint16]bool
+	invalidReceiveLogSizeError string
+)
+
+func init() {
```

Comment:
I would generate `allowedReceiveLogSizes` in `NewReceiveLog` I don't think that is going to be a hot spot.

With `invalidReceiveLogSizeError` I would do a static error, and then include the metadata. That way people can do `errors.Is`

```
InvalidReceiveLogSizeError = errors.New("Invalid receive log size")

...
...
return fmt.Errorf("%w: %d is not a valid size", InvalidReceiveLogSizeError, requestedSize)
```